crates/network-scanner/src/lib.rs

```diff
@@ -0,0 +1,14 @@
+pub mod ping;
+
+#[derive(Debug)]
+pub enum NetworkScanError {
+    IoError(std::io::Error),
+    Other(String),
+    ProtocolError(String),
+}
+
+impl From<std::io::Error> for NetworkScanError {
+    fn from(e: std::io::Error) -> Self {
+        NetworkScanError::IoError(e)
+    }
+}
```

Comment:
note: I would be fine with using `anyhow` in this crate, given that everything is getting stringified here. `anyway::Error` is arguably better at keeping the error structured.

Reply:
updated